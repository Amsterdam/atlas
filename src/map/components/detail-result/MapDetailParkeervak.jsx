import React from 'react'
import PropTypes from 'prop-types'

import MapDetailResultItem from './MapDetailResultItem'
import MapDetailResultWrapper from './MapDetailResultWrapper'

const MapDetailParkeervak = ({ panoUrl, item, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={item.id}
    title="Parkeervak"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem label="Straat" value={item.straatnaam} />
      <MapDetailResultItem label="Type" value={item.e_type_desc} />
      <MapDetailResultItem label="Bord" value={item.bord} />
    </ul>
  </MapDetailResultWrapper>
)

MapDetailParkeervak.propTypes = {
  item: PropTypes.shape({}).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired,
}

export default MapDetailParkeervak
