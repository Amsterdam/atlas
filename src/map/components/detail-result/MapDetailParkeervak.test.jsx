import React from 'react';
import { shallow } from 'enzyme';
import MapDetailParkeervak from './MapDetailParkeervak';


describe('MapDetailParkeervak', () => {
  it('should render everything', () => {
    const item = {
      straatnaam: 'Straatnaam',
      type: 'type',
      bord: 'bord'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailParkeervak
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        item={item}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
