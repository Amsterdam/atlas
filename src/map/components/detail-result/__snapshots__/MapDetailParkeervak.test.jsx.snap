// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailParkeervak should render everything 1`] = `
<Connect(MapDetailResultWrapper)
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  title="Parkeervak"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      hasMultiline={false}
      label="Straat"
      value="Straatnaam"
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Type"
      value=""
    />
    <MapDetailResultItem
      hasMultiline={false}
      label="Bord"
      value="bord"
    />
  </ul>
</Connect(MapDetailResultWrapper)>
`;
